src/org/mozilla/javascript/NativeArray.java

```diff
@@ -641,6 +651,17 @@ private static Object jsConstructor(Context cx, Scriptable scope,
         return new NativeArray(len);
     }
 
+    private static Object jsOf(Context cx, Scriptable scope, Scriptable thisObj, Object[] args)
```

Comment:
Is there any reason why this can't be js_of like other stuff in this class? A minor thing I know.

Reply:
no real reason, i have used the pattern that was used for the static methods, because this is more constructor like.

Not really work in progress because my time is limited and a had this around for some time. Made the pull request because this fixes some more cases and breaks no existings.